src/course-unit/add-component/AddComponent.jsx

```diff
@@ -0,0 +1,64 @@
+import PropTypes from 'prop-types';
+import { useSelector } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import { Button } from '@edx/paragon';
+
+import { getCourseSectionVertical } from '../data/selectors';
+import { COMPONENT_ICON_TYPES } from '../constants';
+import ComponentIcon from './ComponentIcon';
+import messages from './messages';
+
+const AddComponent = ({ blockId, handleCreateNewCourseXblock }) => {
+  const navigate = useNavigate();
+  const intl = useIntl();
+  const { componentTemplates } = useSelector(getCourseSectionVertical);
+
+  const handleCreateNewXblock = (type) => () => {
+    switch (type) {
+    case COMPONENT_ICON_TYPES.discussion:
+    case COMPONENT_ICON_TYPES.dragAndDrop:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId });
+      break;
+    case COMPONENT_ICON_TYPES.problem:
+    case COMPONENT_ICON_TYPES.video:
+      handleCreateNewCourseXblock({ type, parentLocator: blockId }, ({ courseKey, locator }) => {
+        navigate(`/course/${courseKey}/editor/${type}/${locator}`);
+      });
+      break;
+    default:
```

Comment:
[Non-blocking] Do we have a plan to handle "advanced" (i.e, custom) XBlocks?

Reply:
Yes, these are just the xblocks that do not require a modal to specify the exact block. (see mock-up)
<img width="453" alt="Screenshot 2024-02-09 at 12 49 54 PM" src="https://github.com/openedx/frontend-app-course-authoring/assets/42981026/2b272833-8ed1-47d7-98d4-28f4c41840ea">